src/ansi-c/parser.y

```diff
@@ -532,15 +532,62 @@ target_list:
         ;
 
 target:
+          deref_target
+        | target '[' array_index ']'
+        { binary($$, $1, $2, ID_range, $3); }
+        | target '[' array_range ']'
+        { binary($$, $1, $2, ID_range, $3); }
+        ;
+
+array_index:
           identifier
-        | '*' target
+        | integer
+        ;
+
+array_range:
+        array_index ',' array_index
```

Comment:
Potential 🙅 

I'm reviewing commit-by-commit so maybe this is explained later on, but is the syntax we're adding here `array[lower_bound, upper_bound]`? In which case this needs to be changed, that is already totally valid syntax in C ([comma operator](https://en.wikipedia.org/wiki/Comma_operator)).

Reply:
Personally, it's my position that this is acceptable.

 Unless I'm mistaken, from the grammar, this particular production derives only from a particular context (`assigns_opt` which part of the contract productions during a function definition), so even if an argument can be made about the overloading of the operator itself being unwanted, it's within a very narrow context, in order to allow for syntax that is congruent to the mathematical notation.

Reply:
@NlightNFotis considering it'd be a pretty trivial change to make this not be a problem anymore (either use a custom operator like `..` or just accept slightly uglier syntax) I can't really agree to that.